components/renku_data_services/notebooks/core_sessions.py

```diff
@@ -558,14 +560,19 @@ async def patch_session(
             image_secret = get_gitlab_image_pull_secret(
                 nb_config, user, image_pull_secret_name, internal_gitlab_user.access_token
             )
-            if image_secret:
-                updated_secrets = [
-                    secret
-                    for secret in (session.spec.imagePullSecrets or [])
-                    if not secret.name.endswith("-image-secret")
-                ]
-                updated_secrets.append(ImagePullSecret(name=image_pull_secret_name, adopt=True))
-                patch.spec.imagePullSecrets = updated_secrets
+
+            if not image_secret:
+                logger.error(f"Failed to create image pull secret for {image_pull_secret_name}")
+                raise errors.ProgrammingError(
+                    message=f"Failed to create image pull secret for {image_pull_secret_name}"
+                )
+
```

Comment:
This message is what the user will see. We can add some information about how they can try to resolve it.


```suggestion
                    message=f"We cannot retrive credentials for your private image {image}. "
                    "In order to resolve this problem, you can try to log out and back in "
                    "and/or check that you still have permissions for the image repository."
```